src/libraries/System.Text.RegularExpressions/gen/Stubs.cs

```diff
@@ -14,12 +14,13 @@ namespace System.Text
 {
     internal static class StringBuilderExtensions
     {
-        public static unsafe StringBuilder Append(this StringBuilder stringBuilder, ReadOnlySpan<char> span)
+        public static StringBuilder Append(this StringBuilder stringBuilder, ReadOnlySpan<char> span)
         {
-            fixed (char* ptr = &MemoryMarshal.GetReference(span))
+            foreach (char c in span)
             {
-                return stringBuilder.Append(ptr, span.Length);
+                stringBuilder.Append(c);
```

Comment:
This is enforcing a whole load of unnecessary checks for every character of generated source, c.f.

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs#L1005-L1018

Reply:
I wonder if we should introduce a common helper that will encapsulate all unsafe things we have to do due to .NETStandard

Reply:
Eirik marked this as resolved so I assumed perf is not important for this one, I'll revert

Reply:
I resolved assuming it was a stale comment, but I guess probably not?

Reply:
Reverted